prow/plank/reconciler.go

```diff
@@ -357,8 +357,54 @@ func (r *reconciler) syncPendingJob(ctx context.Context, pj *prowv1.ProwJob) (*r
 			pj.Status.PodName = pn
 			r.log.WithFields(pjutil.ProwJobFields(pj)).Info("Pod is missing, starting a new pod")
 		}
-	} else {
+	} else if pod.Status.Reason == Evicted {
+		// Pod was evicted.
+		if pj.Spec.ErrorOnEviction {
+			// ErrorOnEviction is enabled, complete the PJ and mark it as
+			// errored.
+			r.log.WithField("error-on-eviction", true).WithFields(pjutil.ProwJobFields(pj)).Info("Pods Node got evicted, fail job.")
+			pj.SetComplete()
+			pj.Status.State = prowv1.ErrorState
+			pj.Status.Description = "Job pod was evicted by the cluster."
+		} else {
+			// ErrorOnEviction is disabled. Delete the pod now and recreate it in
+			// the next resync.
+			r.log.WithFields(pjutil.ProwJobFields(pj)).Info("Pods Node got evicted, deleting & next sync loop will restart pod")
+			client, ok := r.buildClients[pj.ClusterAlias()]
+			if !ok {
+				return nil, fmt.Errorf("evicted pod %s: unknown cluster alias %q", pod.Name, pj.ClusterAlias())
+			}
+			if finalizers := sets.NewString(pod.Finalizers...); finalizers.Has(kubernetesreporterapi.FinalizerName) {
+				// We want the end user to not see this, so we have to remove the finalizer, otherwise the pod hangs
+				oldPod := pod.DeepCopy()
+				pod.Finalizers = finalizers.Delete(kubernetesreporterapi.FinalizerName).UnsortedList()
+				if err := client.Patch(ctx, pod, ctrlruntimeclient.MergeFrom(oldPod)); err != nil {
+					return nil, fmt.Errorf("failed to patch pod trying to remove %s finalizer: %w", kubernetesreporterapi.FinalizerName, err)
+				}
+			}
+			r.log.WithField("name", pj.ObjectMeta.Name).Debug("Delete Pod.")
+			return nil, ctrlruntimeclient.IgnoreNotFound(client.Delete(ctx, pod))
+		}
+	} else if pod.DeletionTimestamp != nil && pod.Status.Reason == "NodeLost" {
```

Comment:
You can use [NodeUnreachablePodReason](https://pkg.go.dev/k8s.io/kubernetes/pkg/util/node#pkg-constants)

Reply:
this was not introduced in this PR, but I think this is a great idea. Added another commit as there are some go mod change